src/main/java/com/api/csis3275_backend/web/SignInController.java

```diff
@@ -0,0 +1,54 @@
+package com.api.csis3275_backend.web;
+
+import com.api.csis3275_backend.entities.Administrator;
+import com.api.csis3275_backend.entities.Tourist;
+import com.api.csis3275_backend.entities.TravelBuddy;
+import com.api.csis3275_backend.repositories.AdminRepository;
+import com.api.csis3275_backend.repositories.TravelBuddyRepository;
+import com.api.csis3275_backend.repositories.TouristRepository;
+import com.api.csis3275_backend.service.SignInService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@AllArgsConstructor
+@RestController
+public class SignInController {
+
+    AdminRepository adminRepository;
+    TravelBuddyRepository travelBuddyRepository;
+    TouristRepository touristRepository;
+    @PostMapping(path = "/api/signIn")
+    public ResponseEntity<?> signIn(@RequestBody SignInService signInRequest) {
+
+        try {
+            Optional<Administrator> adminData = adminRepository.findByUsername(signInRequest.getUsername());
+            Optional<TravelBuddy> travelBuddyData = travelBuddyRepository.findByUsername(signInRequest.getUsername());
```

Comment:
is this method in travelBuddy repo pushed?

Reply:
> is this method in travelBuddy repo pushed?
nope lol, i just pushed it